.github/workflows/chekc-fork-branch.yaml

```diff
@@ -0,0 +1,22 @@
+name: Check Fork Branch
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Echo Branch Configuration
+        id: branch_configuration
+        run: |
+          echo "BRANCH_NAME: ${{github.head_ref}}"
+          echo "REPO: ${{github.event.pull_request.head.repo.full_name}}"
+          echo "BASE_REF: ${{github.event.pull_request.base.ref}}"
+      - name: Check fork branch
+        if: github.event.pull_request.head.repo.full_name != github.repository && github.head_ref == 'main'
```

Comment:
I don't know my Github action foo very well, but aren't conditions normally expressed in `${{ }}`.

Reply:
At least `if` statement can go without curly brackets, I saw it many times in our current yaml files